### PR TITLE
Allow ctrl + backspace in the editor.

### PR DESCRIPTION
**Pull request by MatiRg**
Deleting words in Edit by using Ctrl+Delete and Ctrl+Backspace.
Issue: https://github.com/colobot/colobot/issues/787

### CODE REVIEW

**Comment by tomangelo2** [comment]
Few issues I've found.
1. If cursor "touches" second word (for example: lorem |ipsum, | is a cursor) and you press Ctrl+Backspace, second word (ipsum) is deleted instead of first one (lorem). Works correctly for Ctrl+Delete.
2. Ctrl+Backspacing multiple words leave all spaces that were between these words.
3. Both deleting and backspacing ignores punctuation marks (commas, dots, dashes, etc).

**Comment by MatiRg** [comment]
1. Overlooked that.
2.So I will make all whitespaces erased
3.I will add special characters then

**Comment by MatiRg** [comment]
Something like this?

**Comment by tomangelo2** [comment]
These issues are fixed now, but I think that one space after deleting would be useful, now nearby words are merged into one. Also I noticed that if cursor is between word and comma (lorem|, ipsum) and press Ctrl+Delete, then lorem is deleted with comma, instead of only comma, which is default behavior in most editors.

**Comment by MatiRg** [comment]
Ok, if You want this I will do it, tomorrow.

**Comment by MatiRg** [comment]
I never used before deleting entire word in any editor, but I looked how in Notepad++ this is working nad I`am will do something like that.

**Comment by MatiRg** [comment]
of course it wont work I just synced my local files with repository :)

**Comment by MatiRg** [comment]
ok

**Comment by HybridDog** [comment]
geany supports ctrl+backspace and lots of others, e.g. ctrl+b to jump to the opposite bracket
https://github.com/geany/geany

File: src/ui/controls/edit.cpp [hunk 1]
```diff
@@ -84,7 +84,17 @@ bool IsSep(int character)
     return !IsWord(character);
 }
 
+bool IsBreaker(char c)
+{
+    return ( c == '.'  || c == '{' || c == '}' ||
+             c == ';' || c == ':' || c == '[' || c == ']' ||
+             c == '(' || c == ')' || c == '=' );
```
Comment (HybridDog) [inline src/ui/controls/edit.cpp]: l think = shouldn't be a breaker, +,-,/ etc are also no breakers and using = without spaces is bad code style, isn't it?

Comment (MatiRg) [inline src/ui/controls/edit.cpp]: Well I would add also +-*\%,? but I`am not sure how Colobot Team want to see it.

**Comment by MatiRg** [comment]
Maybe we should expand delimiter list to: " ' , < > + - * / % ?

**Comment by HybridDog** [comment]
MatiRg, no, by default a space is after , + - * / % ? < >
if the space is not present it can be considered bad code style

**Comment by HybridDog** [comment]
krzys-h, l can't see the problem on your gif due to low image quality.

**Comment by krzys-h** [comment]
```C++
extern void object::New()
{
	object item = radar(Titanium|);
}
```
| means where you should put the cursor. Then do ctrl+backspace and you'll be left with:
```C++
extern void object::New()
{
	object item = radar(|m);
}
```
It works correctly if you put the cursor anywhere else inside `Titanium`

**Pull request merged**